# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerDebugger.cpp
# Commit: 03b025c559e7
# Full Hash: 03b025c559e74975d0b188df43ab2d834f70b8ff
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2021-04-10 03:36:58
# Description:
#   Bug 1701190 - Make sure that a current window context exists. r=dom-worker-reviewers,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D111108
# ==============================================================================

diff -r 31e548229211 -r 03b025c559e7 dom/workers/WorkerDebugger.cpp
--- a/dom/workers/WorkerDebugger.cpp	Fri Apr 09 15:05:43 2021 +0300
+++ b/dom/workers/WorkerDebugger.cpp	Fri Apr 09 12:03:52 2021 +0000
@@ -508,7 +508,7 @@
     BrowsingContext* context = win->GetBrowsingContext();
     if (context) {
       top = context->Top();
-      if (top) {
+      if (top && top->GetCurrentWindowContext()) {
         windowID = top->GetCurrentWindowContext()->OuterWindowId();
         isTopLevel = context->IsTop();
       }
